app/src/main/java/de/westnordost/streetcomplete/data/user/OAuthStore.kt

```diff
@@ -4,17 +4,14 @@ import android.content.SharedPreferences
 import androidx.core.content.edit
 import de.westnordost.streetcomplete.Prefs
 import oauth.signpost.OAuthConsumer
-import javax.inject.Inject
-import javax.inject.Provider
 
 /** Manages saving and loading OAuthConsumer persistently  */
-class OAuthStore @Inject constructor(
+class OAuthStore(
     private val prefs: SharedPreferences,
-    private val oAuthConsumerProvider: Provider<OAuthConsumer>
 ) {
     var oAuthConsumer: OAuthConsumer?
         get() {
-            val result = oAuthConsumerProvider.get()
+            val result = okHttpOAuthConsumer()
```

Comment:
Hmm, you are not using Koin at all here to "inject" this. I can't imagine such a case (replacement for `Provider<T>`) hasn't been considered in Koin.

There should be a `factory<OAuthConsumer> { OkHttpOAuthConsumer(CONSUMER_KEY, CONSUMER_SECRET) }` in the `userModule`, right?
Isn't there a public method on the interface of `userModule` that you can call to get an instance of `OAuthConsumer`? Something like `userModule.get<OAuthConsumer>()`?

Reply:
Resolved in [2324a3e](https://github.com/streetcomplete/StreetComplete/pull/3692/commits/2324a3e9bc18e6f6928a9c3e80ae34712c580db5) via the use of [KoinComponent](https://insert-koin.io/docs/reference/koin-core/koin-component/).